lib/base_service.js

```diff
@@ -390,9 +389,15 @@ class BaseService {
                 config.logging.name = name;
             }
 
-            // And the statsd client
-            if (!config.metrics.name) {
-                config.metrics.name = name;
+            // And metrics clients
+            if (!Array.isArray(config.metrics)) {
```

Comment:
This whole block should be:

```js
if (!Array.isArray(config.metrics)) {
  config.metrics = config.metrics || {};
  config.metrics = [config.metrics];
}
config.metrics.forEach(o => o.name = o.name || name);
```

Also, no need for the `console.log()` since in general we support passing in objects whenever single-item arrays are allowed.